src/test/mockFactory.ts

```diff
@@ -413,6 +413,113 @@ export const createTestHldAzurePipelinesYaml = (
+          `pr_id=$(echo $message | grep -oE '[0-9]+' | head -1 | sed -e 's/^0\\+//')`,
+          `./spk/spk deployment create -n $(INTROSPECTION_ACCOUNT_NAME) -k $(INTROSPECTION_ACCOUNT_KEY) -t $(INTROSPECTION_TABLE_NAME) -p $(INTROSPECTION_PARTITION_KEY) --p3 $(Build.BuildId) --hld-commit-id $commitId --pr $pr_id`,
+          `else`,
+          `./spk/spk deployment create -n $(INTROSPECTION_ACCOUNT_NAME) -k $(INTROSPECTION_ACCOUNT_KEY) -t $(INTROSPECTION_TABLE_NAME) -p $(INTROSPECTION_PARTITION_KEY) --p3 $(Build.BuildId) --hld-commit-id $commitId`,
+          `fi`,
+        ]),
+        displayName:
+          "If configured, update manifest pipeline details in Spektate db before manifest generation",
+        condition:
+          "and(ne(variables['INTROSPECTION_ACCOUNT_NAME'], ''), ne(variables['INTROSPECTION_ACCOUNT_KEY'], ''),ne(variables['INTROSPECTION_TABLE_NAME'], ''),ne(variables['INTROSPECTION_PARTITION_KEY'], ''))",
+      },
+      {
+        task: "ShellScript@2",
+        displayName: "Validate fabrikate definitions",
+        inputs: {
+          scriptPath: "build.sh",
+        },
+        condition: `eq(variables['Build.Reason'], 'PullRequest')`,
+        env: {
+          VERIFY_ONLY: 1,
+        },
+      },
+      {
+        task: "ShellScript@2",
+        displayName:
+          "Transform fabrikate definitions and publish to YAML manifests to repo",
+        inputs: {
+          scriptPath: "build.sh",
+        },
+        condition: `ne(variables['Build.Reason'], 'PullRequest')`,
+        env: {
+          ACCESS_TOKEN_SECRET: "$(PAT)",
+          COMMIT_MESSAGE: "$(Build.SourceVersionMessage)",
+          REPO: "$(MANIFEST_REPO)",
+          BRANCH_NAME: "$(Build.SourceBranchName)",
+        },
+      },
+      {
+        script: generateYamlScript([
+          `. ./build.sh --source-only`,
+          `cd "$HOME"/\${MANIFEST_REPO##*/}`,
+          `latest_commit=$(git rev-parse --short HEAD)`,
+          `url=$(git remote --verbose | grep origin | grep fetch | cut -f2 | cut -d' ' -f1)`,
+          `repourl=\${url##*@}`,
+          `get_spk_version`,
+          `download_spk`,
+          `./spk/spk deployment create -n $(INTROSPECTION_ACCOUNT_NAME) -k $(INTROSPECTION_ACCOUNT_KEY) -t $(INTROSPECTION_TABLE_NAME) -p $(INTROSPECTION_PARTITION_KEY) --p3 $(Build.BuildId) --manifest-commit-id $latest_commit --repository $repourl`,
+        ]),
+        displayName:
+          "If configured, update manifest pipeline details in Spektate db after manifest generation",
+        condition:
+          "and(ne(variables['INTROSPECTION_ACCOUNT_NAME'], ''), ne(variables['INTROSPECTION_ACCOUNT_KEY'], ''),ne(variables['INTROSPECTION_TABLE_NAME'], ''),ne(variables['INTROSPECTION_PARTITION_KEY'], ''))",
+      },
+    ],
+  };
+
+  return asString
+    ? yaml.safeDump(data, { lineWidth: Number.MAX_SAFE_INTEGER })
+    : data;
+};
+
+export const createTestHldAzurePipelinesYamlWithVariableGroup = (
```

Comment:
Wondering if we can reuse the code in `createTestHldAzurePipelinesYaml` for this `createTestHldAzurePipelinesYamlWithVariableGroup`, since the only difference between these two functions as far as i am understanding, is just the variable group?